### PR TITLE
[CSL-802] Add quizzes to node

### PR DESCRIPTION
**Pull request by stanlp1**

### CODE REVIEW

File: src/modules/quizzes.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+   * Retrieve next quiz from api
+   *
+   * @function getNextQuiz
+   * @description Retrieve next quiz from Constructor.io API
+   * @param {string} id - The id of the quiz
+   * @param {string} [parameters] - Additional parameters to refine result set
+   * @param {string} [parameters.section] - Section for customer's product catalog (optional)
+   * @param {array} [parameters.a] - An array for answers in the format [[1,2],[1]]
+   * @param {string} [parameters.version_id] - Specific version id for the quiz.
+   * @param {object} [userParameters] - Parameters relevant to the user request
```
Comment (stanlp1) [inline src/modules/quizzes.js:77]: Search and browse had userParameters like below so I added some of them here. I'm not sure if these endpoints will be tracked on a user basis, or if I should remove these parameters.

`   * @param {object} [userParameters] - Parameters relevant to the user request
   * @param {number} [userParameters.sessionId] - Session ID, utilized to personalize results
   * @param {number} [userParameters.clientId] - Client ID, utilized to personalize results
   * @param {string} [userParameters.userId] - User ID, utilized to personalize results
   * @param {string} [userParameters.segments] - User segments
   * @param {string} [userParameters.testCells] - User test cells
   * @param {string} [userParameters.userIp] - Origin user IP, from client
   * @param {string} [userParameters.userAgent] - Origin user agent, from client`

**Comment by sblaurock** [comment]
@stanlp1 do we think it'd be possible to migrate some of the changes we made in the commit range here into this PR? https://github.com/Constructor-io/constructorio-client-javascript/compare/3a99b4a...cb4ae4f

bit more detailed testing and some cleanup. if it's a big burden let me know and I can take a shot at it

**Comment by stanlp1** [comment]
> @stanlp1 do we think it'd be possible to migrate some of the changes we made in the commit range here into this PR? [Constructor-io/constructorio-client-javascript@3a99b4a...cb4ae4f](https://github.com/Constructor-io/constructorio-client-javascript/compare/3a99b4a...cb4ae4f)
> 
> bit more detailed testing and some cleanup. if it's a big burden let me know and I can take a shot at it

👍 Yep, can take care of that.

**Review by sblaurock — APPROVED** [review]
Made a few small updates - should be ready to go.

File: src/modules/quizzes.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      queryParams.section = section;
+    }
+
+    // Pull version_id from parameters
+    if (versionId) {
+      queryParams.version_id = versionId;
+    }
+
+    // Pull a (answers) from parameters and transform
+    if (a) {
```
Comment (jjl014) [inline src/modules/quizzes.js:60]: Let's change from `a` => `answers` all where everywhere

**Review by sblaurock — CHANGES_REQUESTED** [review]
`a` => `answers`

File: src/modules/quizzes.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/quiz/using_quizzes/#answering-a-quiz
+   * @example
+   * constructorio.quizzes.getNextQuestion('quizId', {
+   *    a: [[1,2],[1]],
+   *    section: '123',
+   *    versionId: '123'
+   * });
+   */
+  getNextQuestion(quizId, parameters, userParameters = {}, networkParameters = {}) {
```
Comment (jjl014) [inline src/modules/quizzes.js:114]: The votes are in! `getNextQuestion` will be changed to `getQuizNextQuestion` 👍

File: src/modules/quizzes.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
+/* eslint-disable object-curly-newline, no-underscore-dangle */ const qs = require('qs');
```
Comment (esezen) [inline src/modules/quizzes.js:1]: qs got moved to the first line somehow 🤔

File: src/modules/quizzes.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/quiz/using_quizzes/#answering-a-quiz
+   * @example
+   * constructorio.quizzes.getNextQuizQuestion('quizId', {
+   *    answers: [[1,2],[1]],
+   *    section: '123',
+   *    versionId: '123'
+   * });
+   */
+  getNextQuizQuestion(quizId, parameters, userParameters = {}, networkParameters = {}) {
```
Comment (esezen) [inline src/modules/quizzes.js:114]: Based on Jimmy's comment, should this be `getQuizNextQuestion`?

File: src/modules/quizzes.js [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+   * @param {string} [userParameters.userId] - User ID, utilized to personalize results
+   * @param {string} [userParameters.segments] - User segments
+   * @param {string} [userParameters.userIp] - Origin user IP, from client
+   * @param {string} [userParameters.userAgent] - Origin user agent, from client
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/quiz/using_quizzes/#answering-a-quiz
+   * @example
+   * constructorio.quizzes.getNextQuizQuestion('quizId', {
```
Comment (esezen) [inline src/modules/quizzes.js:108]: Based on Jimmy's comment, should this be getQuizNextQuestion?

File: src/modules/quizzes.js [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+ */
+class Quizzes {
+  constructor(options) {
+    this.options = options || {};
+  }
+
+  /**
+   * Retrieve quiz question from API
+   *
+   * @function getNextQuizQuestion
```
Comment (esezen) [inline src/modules/quizzes.js:89]: Based on Jimmy's comment, should this be getQuizNextQuestion?

**Comment by esezen** [comment]
Left two small comments rest LGTM!

**Review by esezen — APPROVED** [review]
LGTM! Let's get a final review from @jjl014 and 🚢 this

**Review by jjl014 — APPROVED** [review]
This looks solid to me! Thank you for making all of the changes. You are the best! 😄

**Pull request merged**